### ISSUE TITLE
Source encoding while running tests

### ISSUE THREAD

**Issue by vitalyster** [Microsoft/vscode-java-test#323]
How to specify source encoding? Is it possible to read encoding settings from gradle/maven project (at least when their extensions are installed)?
Or better to force UTF-8 and not default charset under Windows.

**Comment by jdneo** [comment]
Hi, @vitalyster,

Will this comment work for you? https://github.com/redhat-developer/vscode-java/issues/479#issuecomment-380652810

**Comment by vitalyster** [comment]
No, this file contains only
```
connection.project.dir=
eclipse.preferences.version=1
```
for root project and 
```
connection.project.dir=..
eclipse.preferences.version=1
```
for subprojects. 
I have `compileJava.options.encoding` and `compileTestJava.options.encoding` set in desired subproject.
Maybe import does not work well with subprojects.

**Comment by jdneo** [comment]
@vitalyster will the solution in this issue help you with it?

**Comment by vitalyster** [comment]
@jdneo not sure I'm understand your question: there is no solution in this issue

**Comment by jdneo** [comment]
@vitalyster My fault, I forgot to add the link, sorry for the confusion. 😄 

https://github.com/Microsoft/vscode-java-test/issues/37. This is a link I found might be similar to your issue. Would be great if you would like to have a try.

**Comment by jdneo** [comment]
This issue should be fixed in #654.

So we have several ways to set the file encoding when running tests:

1. By default is `utf-8`.
2. The encoding set in the setting `files.encoding`
3. The encoding set in the test configuration. for example:
```json
"java.test.config": {
    "vmargs": ["-Dfile.encoding=Cp1252"]
},
```

The higher ones will be overridden by the lower ones.
